flang/include/flang/Lower/Support/BoxValue.h

```diff
@@ -451,6 +451,15 @@ class ExtendedValue : public details::matcher<ExtendedValue> {
                  [](const auto &box) -> unsigned { return box.rank(); });
   }
 
+  bool hasConstantShape() const {
```

Comment:
I think there are other "constant shape" sorts of queries that work on the type. `hasDynamicSize`, for example.

Reply:
Thanks, using `hasDynamicSize` directly actually removes the need for this helper. I removed it.